README.md

```diff
@@ -90,17 +97,341 @@ for page transitions, and scrollable content within each page.
+          sabGradientColor: _darkSabGradientColor,
+          mainContentScrollPhysics: ClampingScrollPhysics(),
+        ),
+      ],
+    ),
+    home: Scaffold(
+      body: Builder(
+        builder: (context) {
+          return Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Row(...),
+              ElevatedButton(
+                onPressed: () {
+                  WoltModalSheet.show<void>(
+                    context: context,
+                    pageListBuilder: (modalSheetContext) {
+                      final textTheme = Theme.of(context).textTheme;
+                      return [
+                        page1(modalSheetContext, textTheme),
+                        page2(modalSheetContext, textTheme),
+                      ];
+                    },
+                    modalTypeBuilder: (context) {
+                      final size = MediaQuery.sizeOf(context).width;
+                      if (size < _pageBreakpoint) {
+                        return WoltModalType.bottomSheet();
+                      } else {
+                        return WoltModalType.dialog();
+                      }
+                    },
+                    onModalDismissedWithBarrierTap: () {
+                      debugPrint('Closed modal sheet with barrier tap');
+                      Navigator.of(context).pop();
+                    },
+                  );
+                },
+                child: const SizedBox(
+                  height: _buttonHeight,
+                  width: _buttonWidth,
+                  child: Center(child: Text('Show Modal Sheet')),
+                ),
+              ),
+            ],
+          );
+        },
+      ),
+    ),
+  );
+}
+```
+
+The example project above will create the following:
+</br>
+</br>
+
+![Example app](https://github.com/woltapp/wolt_modal_sheet/blob/main/doc/example_wms_demo.gif?raw=true)
+
+### Playground app with imperative navigation
+
+The [playground](./playground/) app demonstrates how to imperatively show the
+modal sheet. The purpose of this module is to play and experiment with various
+use cases.
+
+### Playground app with declarative navigation
+
+The [playground_navigator2](./playground_navigator2/) has the same content with
+the [playground](./playground/) app but the modal sheet is shown using Navigator
+2.0 (Router API) in a declarative way.
+
+### Coffee maker app for state management example
+
+Finally, the [coffee_maker](./coffee_maker/) app demonstrates how to manage the
+state among the page components with an opinionated use of the Provider state
+management library.
+
+The code snippet demonstrates how to decorate the modal sheet with a change
+notifier provider so that the page components can be rebuilt according to the
+current state:
+
+```dart
+  void _onCoffeeOrderSelectedInAddWaterState(
+      BuildContext context, String coffeeOrderId) {
+    final model = context.read<StoreOnlineViewModel>();
```

Comment:
let's call this `viewModel` instead of model. It was a mistake.

Reply:
if this is different in the codebase, let's keep this for now and we will update the code snippets alongside with the sample apps.